bin/theano_nose.py

```diff
@@ -204,6 +205,8 @@ def main():
     if '--help' in sys.argv or '-h' in sys.argv:
         help()
     else:
+        warnings.simplefilter("default")  # Enable warnings before importing theano
+        os.environ["PYTHONWARNINGS"] = "default"  # Also affect subprocesses
```

Comment:
I think this should not overwrite that variable if it is already defined by the user.

Reply:
Fixed in https://github.com/Theano/Theano/pull/6760.

Reply:
This is a good idea. Perhaps the ideal would be to bump `simplefilter()` and `PYTHONWARNINGS` up to `default` if they're weaker, thus preserving stronger filters, but there doesn't seem to be an API to find out what the warnings filter contains.

So it could be worth moving the `warnings.simplefilter("default")` call into the same `if` statement, that is, set both only if the environment variable was unset.